CMakeLists.txt

```diff
@@ -855,43 +863,61 @@ set(devilutionx_assets
   ui_art/mainmenuw.pcx
   ui_art/supportw.pcx)
 
-
-# Copy assets to the build assets subdirectory. This serves two purposes:
-# - If smpq is installed, devilutionx.mpq is built from these files.
-# - If smpq is not installed, the game will load the assets directly from this directoy.
-foreach(asset_file ${devilutionx_assets})
-  set(src "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/resources/assets/${asset_file}")
-  set(dst "${DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY}/${asset_file}")
-  list(APPEND DEVILUTIONX_MPQ_FILES "${asset_file}")
-  list(APPEND DEVILUTIONX_OUTPUT_ASSETS_FILES "${dst}")
-  add_custom_command(
-    COMMENT "Copying ${asset_file}"
-    OUTPUT "${dst}"
-    DEPENDS "${src}"
-    COMMAND ${CMAKE_COMMAND} -E copy "${src}" "${dst}"
-    VERBATIM)
-endforeach()
-if (Gettext_FOUND)
-  foreach(lang ${devilutionx_langs})
-    list(APPEND DEVILUTIONX_MPQ_FILES "${lang}.gmo")
-  endforeach()
+if(VIRTUAL_GAMEPAD)
+  list(APPEND devilutionx_assets
+    ui_art/button.png
+    ui_art/directions2.png
+    ui_art/directions.png
+    ui_art/menu-levelup.png
+    ui_art/menu.png)
 endif()
 
-if(BUILD_ASSETS_MPQ)
-  set(DEVILUTIONX_MPQ "${CMAKE_CURRENT_BINARY_DIR}/devilutionx.mpq")
-  add_custom_command(
-    COMMENT "Building devilutionx.mpq"
-    OUTPUT "${DEVILUTIONX_MPQ}"
-    COMMAND ${CMAKE_COMMAND} -E remove -f "${DEVILUTIONX_MPQ}"
-    COMMAND ${SMPQ} -M 1 -C PKWARE -c "${DEVILUTIONX_MPQ}" ${DEVILUTIONX_MPQ_FILES}
-    WORKING_DIRECTORY "${DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY}"
-    DEPENDS ${DEVILUTIONX_OUTPUT_ASSETS_FILES} ${devilutionx_lang_targets} ${devilutionx_lang_files}
-    VERBATIM)
-  add_custom_target(devilutionx_mpq DEPENDS "${DEVILUTIONX_MPQ}")
-  add_dependencies(${BIN_TARGET} devilutionx_mpq)
+if(APPLE)
+  foreach(asset_file ${devilutionx_assets})
+    set(src "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/resources/assets/${asset_file}")
+    set_source_files_properties("${src}" PROPERTIES
+      MACOSX_PACKAGE_LOCATION Resources/assets

```

Comment:
This isn't going to work since the game will be looking for the files in the subfoders `data`, `ui_art`, `fonts`, `gendata`, not `assets`.

Reply:
Oh, I finally understand what your regexp was doing

Reply:
I can see it being confusing with android having them in `./asset`s, apple using `./resources` an we have them in `./resources/assets` :D

Reply:
OK this should be good now, please give it a try!

Reply:
Oh I didn't think get_filename_component would work with out a file existing in the given path :o